mrp_bom_widget_section_and_note_one2many/models/mrp_bom_line.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (C) 2022 - Today: GRAP (http://www.grap.coop)
+# @author: BADEP
+# @author: Quentin DUPONT (quentin.dupont@grap.coop)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class MrpBomLine(models.Model):
+    _inherit = "mrp.bom.line"
+
+    # To handle bom_line section & note, we have to make these fiels not required
+    product_id = fields.Many2one(required=False)
+    product_qty = fields.Float(required=False)
```

Comment:
I guess you should then add a constraint that these fields are at least present when you are not entering either section or note ?

As done here : https://github.com/odoo/odoo/blob/f31fcc7e72e8e1c73b7f9b38fdb41a2df9e529a2/addons/sale/models/sale.py#L1315-L1322

Also you end up with product_qty = 1 since this is the default for this field, so either you should set it to False when writing a new section / note, or you can leave it required.

Reply:
Thanks fo your review ! I agree with you !

Reply:
I let product_qty required
Remove product_id required on models but add sql_constraints + view also requires it